library/ns1_record.py

```diff
@@ -450,17 +500,68 @@ def update(self, zone, record):
                     changed = True
                     args[key] = input_data
 
+        # create a new copy of the previously sanitized dict that will be updated with chaning args to support --diff
+        after_changes = record_data.copy()
+        for k, v in args.items():
+            if after_changes[k] is not v:
```

Comment:
the `is not` comparison makes me a bit nervous, what's the reason we're not `==` here?

Reply:
Totally understand the unease here since the two operators work differently. The idea behind using `is` instead of the equality operator is for speed. The decision was mostly rooted in knowing args was already being generated as a difference in parameters.  Since `v` will be part of a different object it was counted on in this expression that the identity would never match up anyways. No issues with changing over the operator since the supposed speed boost is probably a drop in the bucket anyways. It might make things more readable for others in the future who have to get up to speed on the logic as well. That being said are you specifically looking for `==` to be used or can I do a direct replacement of `is not` for `!=`?

Reply:
Yeah, it's just the `is not` vs `!=` that makes me nervous. Especially when the things we're comparing can be objects, and nested. My concern is that this might seem to work, but be relying on (for example) that we were doing shallow copies, or relying on the current shape of data.
I'm not saying it's wrong - more that if i see deep copies and `==/!=` comparisons and things working as expected, I'm less worried about unexpected behavior. If there's good reason to do it this way,  a comment explaining that it's intentional (and why) would be appropriate, I think

Reply:
Hey @rupa since we are doing a comparison here and not a calculation I believe` is not` is the appropriate use and not sure it should be documented as expected, I could be wrong but using` is not` vs `!=` when comparing objects is best practice and we shouldn't need to document best practice. It's an `identity test `vs an `equality test`, but maybe I'm wrong in my thinking, if so please teach me :).

Reply:
yeah im not sure we want identity comparisons here. it makes me nervous that shallow/deep copy might affect how it works, although they behave the same in this silly little check
```
>>> x = {'a': {'b': {'c': 'd'}}}
>>> y = x.copy()
>>> x == y
True
>>> x is y
False
>>> y = deepcopy(x)
>>> y == x
True
>>> y is x
False
```
in any event, I'm not 100 saying this is wrong, but it's not what I'd expect, so I mostly just want to make sure perfectly sure we're going to have the behavior we want.